optaplanner-core/src/main/java/org/optaplanner/core/config/solver/termination/TerminationConfig.java

```diff
@@ -562,4 +563,12 @@ public TerminationConfig copyConfig() {
         return new TerminationConfig().inherit(this);
     }
 
+    @Override
+    public void visitReferencedClasses(Consumer<Class<?>> classVisitor) {
+        classVisitor.accept(terminationClass);
+        if (terminationConfigList != null) {
+            terminationConfigList.forEach(tc -> tc.visitReferencedClasses(classVisitor));
+        }
+    }
```

Comment:
The more I think about this API, the more I come to the conclusion that this method should be hidden from users.
Even though `core.config` technically isn't public API, we still treat it like one, as people will be using it. In my opinion, they should not be able to access this method.

Reply:
It does need to be accessible from optaplanner-quarkus deployment though. Making it protected will not work (different classloader + different package => not accessible). The only way I see to make this possible is to make this protected (cannot be private since it need to be overriden), and use reflection to access it from Quarkus.

Reply:
It also need to be accessible from benchmark since some benchmark config override AbstractConfig

Reply:
Yeah, let's keep it as is. Anything else is too much hassle for too little benefit.